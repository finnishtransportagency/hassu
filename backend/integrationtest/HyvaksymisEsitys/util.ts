import { config } from "../../src/config";
import { getS3Client } from "../../src/aws/client";
import { DeleteObjectsCommand, ListObjectsV2Command, ListObjectsV2CommandOutput, PutObjectCommand } from "@aws-sdk/client-s3";
import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
import { adaptFileName } from "../../src/tiedostot/paths";
import { getAxios } from "../../src/aws/monitoring";
import FormData from "form-data";

export async function insertYllapitoFileToS3(pathInS3: string) {
  if (pathInS3.includes(config.yllapitoBucketName)) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  const buffer = Buffer.from("Test");
  await getS3Client().send(
    new PutObjectCommand({
      Body: buffer,
      Bucket: config.yllapitoBucketName,
      Key: pathInS3,
    })
  );
}

export async function insertUploadFileToS3(uuid: string, filename: string) {
  const fileNameWithPath = `${uuid}/${adaptFileName(filename)}`;
  const presignedPost = await createPresignedPost(getS3Client(), {
    Bucket: config.uploadBucketName,
    Key: fileNameWithPath,
    Expires: 600,
  });
  const fields = presignedPost.fields;
  const form = new FormData();
  Object.keys(fields).forEach((key) => {
    form.append(key, fields[key]);
  });
  form.append("file", "Test");
  const axios = getAxios();
  const lengthSync = form.getLengthSync();

  await axios.post(presignedPost.url, form, {
    headers: {
      "Content-Length": lengthSync,
      ...form.getHeaders(),
    },
  });
}

export async function deleteYllapitoFiles(sourcePrefix?: string) {
  if (sourcePrefix && sourcePrefix.includes(config.yllapitoBucketName)) {
    throw new Error("sourcePrefix ei saa sisältää yllapitoBucketNamea");
  }
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: sourcePrefix,
        ContinuationToken,
      })
    );
    await deleteFiles(Contents.map(({ Key }) => Key as string));

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

export async function emptyUploadFiles() {
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.uploadBucketName,
        ContinuationToken,
      })
    );
    await deleteFiles(
      Contents.map(({ Key }) => Key as string),
      config.uploadBucketName
    );

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

async function deleteFiles(paths: string[], bucket: string = config.yllapitoBucketName) {
  if (paths.some((path) => path.includes(config.yllapitoBucketName))) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  await Promise.all(
    // DeleteObjectsCommand takes at most 1000 items
    getChunksOfThousand(paths).map((paths) =>
      getS3Client().send(
        new DeleteObjectsCommand({
          Bucket: bucket,
          Delete: { Objects: paths.map((path) => ({ Key: path })) },
        })
      )
    )
  );
}

function getChunksOfThousand<T>(array: T[]): T[][] {
  const chunks: T[][] = [];
  for (let i = 0; i < array.length; i += 1000) {
    chunks.push(array.slice(i, i + 1000));
  }
  return chunks;
}

export async function getYllapitoFilesUnderPath(pathInS3: string): Promise<string[]> {
  const s3 = getS3Client();
  const files: string[] = [];
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: pathInS3,
        ContinuationToken,
      })
    );
    files.push(...Contents.map(({ Key }) => Key as string));
    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
  return files;
}
