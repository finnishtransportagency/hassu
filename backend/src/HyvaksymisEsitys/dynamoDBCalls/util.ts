import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
import { log } from "../../logger";
import { getDynamoDBDocumentClient } from "../../aws/client";
import { ConditionalCheckFailedException } from "@aws-sdk/client-dynamodb";
import { SimultaneousUpdateError } from "hassu-common/error";

export async function sendUpdateCommandToDynamoDB(params: UpdateCommand): Promise<void> {
  if (log.isLevelEnabled("debug")) {
    log.debug("Updating projekti to Hassu with params", { params });
  }
  try {
    const dynamoDBDocumentClient = getDynamoDBDocumentClient();
    await dynamoDBDocumentClient.send(params);
  } catch (e) {
    if (e instanceof ConditionalCheckFailedException) {
      throw new SimultaneousUpdateError("Projektia on p√§ivitetty tietokannassa. Lataa projekti uudelleen.");
    }
    log.error(e instanceof Error ? e.message : String(e), { params });
    throw e;
  }
}
