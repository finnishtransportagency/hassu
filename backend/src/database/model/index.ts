export * from "./common";
export * from "./projekti";
export * from "./suunnitteluVaihe";
export * from "./nahtavillaoloVaihe";
export * from "./hyvaksymisPaatosVaihe";
