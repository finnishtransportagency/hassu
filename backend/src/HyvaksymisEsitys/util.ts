import { IllegalArgumentError } from "hassu-common/error";
import { DBProjekti } from "../database/model";
import { projektiDatabase } from "../database/projektiDatabase";
import { requirePermissionLuku } from "../user";
import { NykyinenKayttaja } from "hassu-common/graphql/apiModel";

export async function varmistaLukuoikeusJaHaeProjekti(
  oid: string
): Promise<{ projektiInDB: DBProjekti; nykyinenKayttaja: NykyinenKayttaja }> {
  const nykyinenKayttaja = requirePermissionLuku();
  const projektiInDB = await projektiDatabase.loadProjektiByOid(oid);
  if (!projektiInDB) {
    throw new IllegalArgumentError(`Projektia oid:lla ${oid} ei l√∂ydy`);
  }
  return { projektiInDB, nykyinenKayttaja };
}
