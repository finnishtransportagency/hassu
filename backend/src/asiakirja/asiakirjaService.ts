import { AsiakirjaTyyppi, Kieli, PDF, ProjektiTyyppi } from "../../../common/graphql/apiModel";
import { AloitusKuulutusJulkaisu } from "../database/model/projekti";
import { AloitusKuulutus10T } from "./suunnittelunAloitus/aloitusKuulutus10T";
import { AloitusKuulutus10R } from "./suunnittelunAloitus/aloitusKuulutus10R";
import { Ilmoitus12T } from "./suunnittelunAloitus/ilmoitus12T";
import { Ilmoitus12R } from "./suunnittelunAloitus/ilmoitus12R";

interface CreatePdfOptions {
  aloitusKuulutusJulkaisu: AloitusKuulutusJulkaisu;
  asiakirjaTyyppi: AsiakirjaTyyppi;
  kieli: Kieli;
}

enum AsiakirjanMuoto {
  TIE,
  RATA,
}

function determineAsiakirjaMuoto(aloitusKuulutusJulkaisu: AloitusKuulutusJulkaisu): AsiakirjanMuoto | null {
  const tyyppi = aloitusKuulutusJulkaisu.velho.tyyppi;
  const vaylamuoto = aloitusKuulutusJulkaisu.velho?.vaylamuoto;
  if (tyyppi === ProjektiTyyppi.TIE || (tyyppi === ProjektiTyyppi.YLEINEN && vaylamuoto?.includes("tie"))) {
    return AsiakirjanMuoto.TIE;
  } else if (tyyppi === ProjektiTyyppi.RATA || (tyyppi === ProjektiTyyppi.YLEINEN && vaylamuoto?.includes("rata"))) {
    return AsiakirjanMuoto.RATA;
  }
  return null;
}

export class AsiakirjaService {
  createPdf({ asiakirjaTyyppi, aloitusKuulutusJulkaisu, kieli }: CreatePdfOptions): Promise<PDF> {
    let pdf: Promise<PDF>;
    const asiakirjanMuoto = determineAsiakirjaMuoto(aloitusKuulutusJulkaisu);

    switch (asiakirjaTyyppi) {
      case AsiakirjaTyyppi.ALOITUSKUULUTUS:
        switch (asiakirjanMuoto) {
          case AsiakirjanMuoto.TIE:
            pdf = new AloitusKuulutus10T(aloitusKuulutusJulkaisu, kieli).pdf;
            break;
          case AsiakirjanMuoto.RATA:
            pdf = new AloitusKuulutus10R(aloitusKuulutusJulkaisu, kieli).pdf;
            break;
          default:
            throw new Error(
              `Aloituskuulutuspohjaa ei pystytä päättelemään. tyyppi: '${aloitusKuulutusJulkaisu.velho.tyyppi}', vaylamuoto: '${aloitusKuulutusJulkaisu.velho?.vaylamuoto}'`
            );
        }
        break;
      case AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA:
        switch (asiakirjanMuoto) {
          case AsiakirjanMuoto.TIE:
            pdf = new Ilmoitus12T(aloitusKuulutusJulkaisu, kieli).pdf;
            break;
          case AsiakirjanMuoto.RATA:
            pdf = new Ilmoitus12R(aloitusKuulutusJulkaisu, kieli).pdf;
            break;
          default:
            throw new Error(
              `Ilmoituspohjaa ei pystytä päättelemään. tyyppi: '${aloitusKuulutusJulkaisu.velho.tyyppi}', vaylamuoto: '${aloitusKuulutusJulkaisu.velho?.vaylamuoto}'`
            );
        }
        break;
      default:
        throw new Error(`Asiakirjatyyppi ('${asiakirjaTyyppi}') ei ole vielä tuettu`);
    }
    return pdf;
  }
}

export const asiakirjaService = new AsiakirjaService();
