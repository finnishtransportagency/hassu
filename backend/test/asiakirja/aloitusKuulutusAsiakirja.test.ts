import { describe, it } from "mocha";
import { AloituskuulutusPdfOptions } from "../../src/asiakirja/asiakirjaTypes";
import { AsiakirjaTyyppi, Kieli, ProjektiTyyppi, SuunnittelustaVastaavaViranomainen } from "hassu-common/graphql/apiModel";
import { asiakirjaAdapter } from "../../src/handler/asiakirjaAdapter";
import { ProjektiFixture } from "../fixture/projektiFixture";
import { AloitusKuulutusJulkaisu, DBVaylaUser } from "../../src/database/model";
import * as sinon from "sinon";
import { AsiakirjaService } from "../../src/asiakirja/asiakirjaService";
import { expectPDF } from "./asiakirjaTestUtil";
import { assertIsDefined } from "../../src/util/assertions";
import { defaultUnitTestMocks } from "../mocks";
import { S3Mock } from "../aws/awsMock";
import { KaannettavaKieli } from "hassu-common/kaannettavatKielet";

import { expect } from "chai";

const projektiFixture = new ProjektiFixture();

describe("aloitusKuulutusAsiakirja", () => {
  defaultUnitTestMocks();

  new S3Mock(true);

  afterEach(() => sinon.reset());
  after(() => sinon.restore());

  it("should generate kuulutus pdf succesfully SUOMI (tie) (suunnittelusopimus)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, true, ProjektiTyyppi.TIE));
  it("should generate kuulutus pdf succesfully SUOMI (yleis) (suunnittelusopimus)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, true, ProjektiTyyppi.YLEINEN));
  it("should generate kuulutus pdf succesfully SUOMI (tie)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, false, ProjektiTyyppi.TIE));
  it("should generate kuulutus pdf succesfully SUOMI (rata)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, false, ProjektiTyyppi.RATA));
  it("should generate kuulutus pdf succesfully SUOMI (tie, vähäinen menettely)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, false, ProjektiTyyppi.TIE, true));
  it("should generate kuulutus pdf succesfully SUOMI (rata, vähäinen menettely)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ALOITUSKUULUTUS, Kieli.SUOMI, false, ProjektiTyyppi.RATA, true));

  it("should generate ilmoitus pdf succesfully SUOMI (tie) (suunnittelusopimus)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, true, ProjektiTyyppi.TIE));
  it("should generate ilmoitus pdf succesfully SUOMI (yleis) (suunnittelusopimus)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, true, ProjektiTyyppi.YLEINEN));
  it("should generate ilmoitus pdf succesfully SUOMI (tie)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, false, ProjektiTyyppi.TIE));
  it("should generate ilmoitus pdf succesfully SUOMI (rata)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, false, ProjektiTyyppi.RATA));
  it("should generate ilmoitus pdf succesfully SUOMI (tie, vähäinen menettely)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, false, ProjektiTyyppi.TIE, true));
  it("should generate ilmoitus pdf succesfully SUOMI (rata, vähäinen menettely)", async () =>
    doTestGenerateKuulutus(AsiakirjaTyyppi.ILMOITUS_KUULUTUKSESTA, Kieli.SUOMI, false, ProjektiTyyppi.RATA, true));
});

async function doTestGenerateKuulutus(
  asiakirjaTyyppi: AsiakirjaTyyppi,
  kieli: KaannettavaKieli,
  suunnitteluSopimus: boolean,
  projektiTyyppi: ProjektiTyyppi,
  vahainenMenettely?: boolean
): Promise<void> {
  const projekti = projektiFixture.dbProjekti1(); // Suomi+Ruotsi
  assertIsDefined(projekti.velho);
  projekti.velho.tyyppi = projektiTyyppi;
  if (projektiTyyppi == ProjektiTyyppi.RATA) {
    projekti.velho!.suunnittelustaVastaavaViranomainen = SuunnittelustaVastaavaViranomainen.VAYLAVIRASTO;
    projekti.velho.vaylamuoto = ["rata"];
  } else {
    projekti.velho.vaylamuoto = ["tie"];
  }
  const aloitusKuulutusJulkaisu = await asiakirjaAdapter.adaptAloitusKuulutusJulkaisu(projekti);
  if (suunnitteluSopimus) {
    expect(aloitusKuulutusJulkaisu.suunnitteluSopimus).not.to.be.undefined;
  } else {
    aloitusKuulutusJulkaisu.suunnitteluSopimus = undefined;
  }

  await testKuulutusWithLanguage(
    projekti.oid,
    aloitusKuulutusJulkaisu,
    kieli,
    projekti.kayttoOikeudet,
    asiakirjaTyyppi,
    !!vahainenMenettely,
    projektiTyyppi,
    suunnitteluSopimus ? "suunnittelusopimus" : "",
    vahainenMenettely ? "vahainen_menettely" : ""
  );
}

async function testKuulutusWithLanguage(
  oid: string,
  aloitusKuulutusJulkaisu: AloitusKuulutusJulkaisu,
  kieli: KaannettavaKieli,
  kayttoOikeudet: DBVaylaUser[],
  asiakirjaTyyppi: AsiakirjaTyyppi,
  vahainenMenettely: boolean,
  ...description: string[]
): Promise<void> {
  const aloituskuulutusPdfOptions: AloituskuulutusPdfOptions = {
    oid,
    lyhytOsoite: "ABCD",
    aloitusKuulutusJulkaisu,
    asiakirjaTyyppi,
    kieli,
    luonnos: true,
    kayttoOikeudet,
    vahainenMenettely,
  };
  const pdf = await new AsiakirjaService().createAloituskuulutusPdf(aloituskuulutusPdfOptions);
  expect(pdf.sisalto.length).to.be.greaterThan(30000);
  expectPDF("esikatselu_aloituskuulutus_" + description.filter((text) => !!text).join("_") + "_", pdf, asiakirjaTyyppi);
}
