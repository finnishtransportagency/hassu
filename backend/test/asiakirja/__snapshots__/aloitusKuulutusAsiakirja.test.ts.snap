// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aloitusKuulutusAsiakirja should generate ilmoitus pdf succesfully SUOMI (rata) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_RATA__12R Ilmoitus aloituskuulutuksesta.pdf",
  "textContent": "VAYLA/1/2022



Ilmoitus Väyläviraston kuulutuksesta
Testiprojekti 1
Väylävirasto julkaisee kuulutuksen, joka koskee otsikossa mainitun ratasuunnitelman suunnittelun ja maastotöiden aloittamista. Väylävirasto saattaa asian tiedoksi julkisesti kuuluttamalla siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (ratalaki 95 §/LjMTL 108§, HL 62 a §) 
Kuulutus julkaistaan 2.1.2022 osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus



Kuulutus on nähtävillä 2.1.2022 - 1.2.2022 välisen ajan
Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate ilmoitus pdf succesfully SUOMI (tie) (suunnittelusopimus) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_TIE_suunnittelusopimus_T412_1 Ilmoitus aloituskuulutuksesta.pdf",
  "textContent": "ELY/1/2022



Ilmoitus aloituskuulutuksesta
Testiprojekti 1
Nokia, sovittuaan asiasta Uudenmaan ELY-keskuksen kanssa julkaisee kuulutuksen, joka koskee otsikossa mainitun tiesuunnitelman suunnittelun ja maastotöiden aloittamista. Nokia saattaa asian tiedoksi julkisesti kuuluttamalla siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (LjMTL 108§, HL 62 a §)
Kuulutus julkaistaan 2.1.2022 osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus



Kuulutus on nähtävillä 2.1.2022 - 1.2.2022 välisen ajan
Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate ilmoitus pdf succesfully SUOMI (tie) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_TIE__T412_1 Ilmoitus aloituskuulutuksesta.pdf",
  "textContent": "ELY/1/2022



Ilmoitus Uudenmaan ELY-keskuksen kuulutuksesta
Testiprojekti 1
Uudenmaan elinkeino-, liikenne- ja ympäristökeskus (ELY-keskus) julkaisee kuulutuksen, joka koskee otsikossa mainitun tiesuunnitelman suunnittelun ja maastotöiden aloittamista. Uudenmaan ELY-keskus saattaa asian tiedoksi julkisesti kuuluttamalla siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (LjMTL 108§, HL 62 a §) 
Kuulutus julkaistaan 2.1.2022 osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus



Kuulutus on nähtävillä 2.1.2022 - 1.2.2022 välisen ajan
Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate ilmoitus pdf succesfully SUOMI (yleis) (suunnittelusopimus) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_YLEINEN_suunnittelusopimus_12YS Ilmoitus aloituskuulutuksesta.pdf",
  "textContent": "ELY/1/2022



Ilmoitus aloituskuulutuksesta
Testiprojekti 1
Nokia, sovittuaan asiasta Uudenmaan ELY-keskuksen kanssa julkaisee kuulutuksen, joka koskee otsikossa mainitun yleissuunnitelman suunnittelun ja maastotöiden aloittamista. Nokia saattaa asian tiedoksi julkisesti kuuluttamalla siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (LjMTL 108§, HL 62 a §)
Kuulutus julkaistaan 2.1.2022 osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus



Kuulutus on nähtävillä 2.1.2022 - 1.2.2022 välisen ajan
Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate kuulutus pdf succesfully SUOMI (rata) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_RATA__10R Aloituskuulutus.pdf",
  "textContent": "VAYLA/1/2022



Kuulutus suunnittelun aloittamisesta
Testiprojekti 1
Väylävirasto aloittaa otsikon mukaisen ratasuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.
Lorem Ipsum
Väylävirasto on 2.1.2022 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus. Kuulutus pidetään nähtävillä 2.1.2022 - 1.2.2022.



Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä kalenteripäivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen rataverkon haltijalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (ratalaki 9 §).
Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (ratalaki 22 §). 
Väylävirasto käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.vayla.fi/tietosuoja.



Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate kuulutus pdf succesfully SUOMI (tie) (suunnittelusopimus) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_TIE_suunnittelusopimus_T412 Aloituskuulutus.pdf",
  "textContent": "ELY/1/2022



Kuulutus suunnittelun aloittamisesta
Testiprojekti 1
Nokia, sovittuaan asiasta Uudenmaan ELY-keskuksen kanssa, aloittaa otsikon mukaisen tiesuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.
Lorem Ipsum
Nokia on 2.1.2022 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus. Kuulutus pidetään nähtävillä 2.1.2022 - 1.2.2022.



Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä kalenteripäivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen suunnitelmasta vastaavalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (laki liikennejärjestelmästä ja maanteistä LjMTL 16 §).
Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (LjMTL 27 § ja valtioneuvoston asetus maanteistä 3 §). 
Uudenmaan ELY-keskus käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.ely-keskus.fi/tietosuoja.



Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate kuulutus pdf succesfully SUOMI (tie) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_TIE__T412 Aloituskuulutus.pdf",
  "textContent": "ELY/1/2022



Kuulutus suunnittelun aloittamisesta
Testiprojekti 1
Uudenmaan elinkeino-, liikenne- ja ympäristökeskus (ELY-keskus) aloittaa otsikon mukaisen tiesuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.
Lorem Ipsum
Uudenmaan ELY-keskus on 2.1.2022 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus. Kuulutus pidetään nähtävillä 2.1.2022 - 1.2.2022.



Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä kalenteripäivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen suunnitelmasta vastaavalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (laki liikennejärjestelmästä ja maanteistä LjMTL 16 §).
Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (LjMTL 27 § ja valtioneuvoston asetus maanteistä 3 §). 
Uudenmaan ELY-keskus käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.ely-keskus.fi/tietosuoja.



Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;

exports[`aloitusKuulutusAsiakirja should generate kuulutus pdf succesfully SUOMI (yleis) (suunnittelusopimus) 1`] = `
Object {
  "fileName": "esikatselu_aloituskuulutus_YLEINEN_suunnittelusopimus_10YS Aloituskuulutus.pdf",
  "textContent": "ELY/1/2022



Kuulutus suunnittelun aloittamisesta
Testiprojekti 1
Nokia, sovittuaan asiasta Uudenmaan ELY-keskuksen kanssa, aloittaa otsikon mukaisen tiesuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.
Lorem Ipsum
Nokia on 2.1.2022 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa https://localhost:3000/suunnitelma/1/aloituskuulutus. Kuulutus pidetään nähtävillä 2.1.2022 - 1.2.2022.



Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä kalenteripäivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen suunnitelmasta vastaavalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (laki liikennejärjestelmästä ja maanteistä LjMTL 16 §).
Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (LjMTL 27 § ja valtioneuvoston asetus maanteistä 3 §). 
Uudenmaan ELY-keskus käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.ely-keskus.fi/tietosuoja.



Lisätietoja antaa 
Kunta Kuntalainen (Nokia), 
puh. 123456789, Kunta.Kuntalainen(at)vayla.fi 
Eemil Elylainen (Pirkanmaan ELY-keskus), 
puh. 123456789, eemil.elylainen(at)ely.fi 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 
LUONNOS",
}
`;
